build.rs

```diff
@@ -822,24 +849,18 @@ fn ring_core_prefix() -> String {
 
 /// Creates the necessary header file for symbol renaming and returns the path of the
 /// generated include directory.
-fn generate_prefix_symbols(
-    target: &Target,
-    out_dir: &Path,
-    prefix: &str,
-) -> Result<(), std::io::Error> {
+fn generate_prefix_symbols(out_dir: &Path, prefix: &str) -> Result<(), std::io::Error> {
     generate_prefix_symbols_header(out_dir, "prefix_symbols.h", '#', None, prefix)?;
 
-    if target.os == "windows" {
```

Comment:
Confirming for myself: This is no longer needed because this function being called is dependent on `preassemble` being `true`, which is for Windows but not for any other target?

Reply:
It's called for two contexts. This is true for the pregeneration, but it's also called in `build_c_code` which happens when it's used as a dependency.

The difference is that Windows now needs access to *both* because of the `clang` usage.

Reply:
Got it. That makes sense, thank you.